test/jdk/java/net/httpclient/websocket/WebSocketProxyTest.java

```diff
@@ -148,6 +150,35 @@ protected PasswordAuthentication getPasswordAuthentication() {
         };
     }
 
+    record bytes(byte[] bytes) {
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o instanceof bytes other) {
+                return Arrays.equals(bytes(), other.bytes());
+            }
+            return false;
+        }
+        @Override
+        public int hashCode() { return Arrays.hashCode(bytes()); }
+        public String toString() {
+            return "0x" + HexFormat.of()
+                    .withUpperCase()
+                    .formatHex(bytes());
+        }
+    }
+
+    static List<bytes> ofBytes(List<byte[]> bytes) {
+        return bytes.stream().map(bytes::new).toList();
+    }
+
+    static String diagnose(List<byte[]> a, List<byte[]> b) {
+        var actual = ofBytes(a);
+        var expected = ofBytes(b);
+        var problem = actual.equals(expected) ? "match" : "differ";
+        return "%s and %s %s".formatted(actual, expected, problem);
```

Comment:
I would rename "problem" to more neutral "message" or "description". Ditto in the other file.

Reply:
Done.